### PR TITLE
feat: label subagent execution tool rows

### LINKED ISSUE #8639: CLI + extension: show subagent identity on executions wait/view tool rows
Product-sense request from maintainer review (2026-07-16, during the TUI audit sequence):

When a transcript tool row shows an `executions` operation (`wait`, `view`, …) whose execution id corresponds to a **subagent**, the row should gracefully display *who* that is — the subagent's agent name/label (e.g. `executions (wait: reviewer, leanSolver)`) — instead of only opaque execution ids. The same applies to any wait-style surface that lists execution targets.

Sketch:
- The mapping already exists in the child-execution state (`state/childExecutions.ts` / `SubagentListDisplay`) — the renderer needs a lookup from execution id → child label at display time, resolved in the data layer (per the render-time-workarounds rule, resolve upstream where the tool row's display model is built, not in the JSX).
- Natural home: the tool-row display-line builder (`panes/toolRenderers.tsx`), which #8628 (PR 3 of the audit sequence) is about to rework into a styled-line SSOT — this labeling should land on top of that model rather than the current dual JSX/plain pipeline.
- Non-subagent executions (plain background processes) keep their current title.

Related: #8628 (tool-row SSOT), #8629 (subagent navigation convergence).

### PR DESCRIPTION
**Pull request by LionSR**
Closes #8639

## Summary
- add one browser-safe executions summary model shared by the CLI and extension
- derive retained execution-id to agent-label maps from each host existing child-stream state
- render subagent names in wait/view rows while preserving current background-process titles

## Design
Identity resolution happens before JSX/Lit rendering. The CLI keeps one derived projection of its consolidated child-stream store, while the extension derives labels from retained child stream metadata. Both feed the same pure executions summary function into their existing tool display models, so live rows, static transcript geometry, printed output, grouped cards, and timeline cards stay consistent.

## Validation
- npm run format (targeted files)
- npm run compile:fast
- npm run lint (0 errors; one unrelated existing warning)
- npm run typecheck
- npm test (728 files passed, 1 skipped; 6655 tests passed, 4 skipped)
- focused post-rebase suite (6 files, 205 tests)

---

> [!NOTE]
> **Low Risk**
> Display-layer and transcript repaint wiring only; no auth or execution control changes. Main risk is incorrect scrollback repaint or label leakage in edge cases, covered by new tests.
> 
> **Overview**
> **Executions** wait/view tool rows can show **retained subagent names** (e.g. `wait: reviewer, leanSolver`) instead of raw execution IDs, while **background-process** targets keep their existing titles when no subagent label applies.
> 
> A shared **`executionsSubagentSummary`** helper in `@shared/tools/executionsDisplay` drives the header text from tool input plus an `executionId → label` map. The **CLI** derives that map from child-stream state (`subagentExecutionLabels`) and threads **`ExecutionLabels`** through live pane, static scrollback, row budgeting, and printed transcript paths so geometry and text stay aligned. The **extension progress view** builds the same map from retained child stream metadata and passes it into tool formatters and log cards.
> 
> Static transcript **repaint** is keyed off scrollback owner **and** label changes: viewport-switch callbacks are replaced with **`onStaticTranscriptChange`** / **`repaintTranscript`** when the static render identity changes (including late-arriving labels), so finalized rows can refresh without relying only on stream tab switches.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b88141e91b4d18ee81f9f146a484891f51ebe3d7. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/extension/src/progressView/frontend/progressState.ts [hunk 1]
```diff
@@ -263,6 +265,26 @@ const activeIsToolUse$ = new Signal.Computed(() => {
   return state ? isToolUseState(state) : false;
 });
 
+/** Stable identity projection; ordinary log ticks do not rebuild this map. */
+const activeSubagentExecutionLabels$ = new Signal.Computed(
+  (): ExecutionLabels => {
+    const activeStreamInfo = activeStreamInfo$.get();
+    if (!activeStreamInfo) return EMPTY_EXECUTION_LABELS;
+    const children = childStreamsByParent$.get().get(activeStreamInfo.name);
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/extension/src/progressView/frontend/progressState.ts]: Fixed at the ownership boundary in 4b98489d57: the extension now derives one session-wide label projection from every retained agent child stream, rather than looking only under the active tab. This matches the CLI child-state model and lets a subagent tab resolve sibling and nested execution targets. Added an exact regression test with a sibling target while the child tab is active.

**Comment by LionSR** [comment]
@codex review exact head 0565d1f9354e40a454240ebbafbeecabb7d775a7

**Comment by LionSR** [comment]
All current-head CI checks pass and the medium-severity sibling-label finding is resolved. @codex review exact head 0565d1f9354e40a454240ebbafbeecabb7d775a7

File: packages/extension/src/progressView/frontend/progressState.ts [hunk 2]
```diff
@@ -58,6 +59,7 @@ const EMPTY_INQUIRIES: InquiryThreadUpdatedEvent[] = [];
 
 /** Stable empty map returned when no parent has active children. */
 const EMPTY_CHILD_MAP: Map<StreamTabId, StreamTabInfo[]> = new Map();
+const EMPTY_EXECUTION_LABELS: ExecutionLabels = new Map();
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/extension/src/progressView/frontend/progressState.ts]: Fixed in bbfa678db5. The derived signal now always returns the Map instance it owns, including when empty, and EMPTY_LOG_CONTEXT omits the optional label map entirely. LogList creates a per-cache-entry fallback only at its ownership boundary, so no shared mutable Map crosses a module boundary.

**Comment by LionSR** [comment]
@codex review exact head bbfa678db5777d68788967ba96e9ba9cb93160c3

**Comment by LionSR** [comment]
@codex review exact head bbfa678db5777d68788967ba96e9ba9cb93160c3

File: packages/cli/src/chat/tui/state/childExecutions.ts [hunk 3]
```diff
@@ -58,6 +58,21 @@ export const childStreamEntries: Signal.Computed<ChildStreamEntries> = computed(
   () => CHILD_STREAMS.get(),
 );
 
+/** Retained execution-id → human label projection for executions tool rows. */
+export const subagentExecutionLabels: Signal.Computed<
+  ReadonlyMap<string, string>
+> = computed(() => {
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/chat/tui/state/childExecutions.ts:64]: Thanks. I’m keeping the derived Map owner-local here: returning a stable empty Map would reintroduce the shared mutable sentinel that the exact-head Codex review correctly flagged against the repository rule. The cache key already uses the resolved summary content, so cache correctness and reuse do not depend on Map identity. A content-stable signal would add state/comparison complexity without measured evidence, so this stays a documented non-blocking optimization rather than widening this PR.

File: packages/cli/src/chat/tui/panes/StaticConversationTranscript.tsx [hunk 4]
```diff
@@ -556,6 +575,7 @@ export function StaticConversationTranscript({
     scrollbackStreamId,
     sessionMeta,
     streams,
+    subagentExecutionLabels,
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/chat/tui/panes/StaticConversationTranscript.tsx:590]: Confirmed as an accepted property of terminal scrollback, not a second mutable rendering model. The roster fact normally precedes a later executions wait/view and an entry only becomes static after leaving the live window; once Ink flushes it, terminal scrollback is intentionally immutable. Rebuilding already-flushed rows would require terminal replay/clear semantics and materially widen this display-only issue. Live, printed, and not-yet-flushed paths all use the retained session-wide labels.

File: src/shared/tools/executionsDisplay.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    matchedSubagent = true;
+    return label;
+  });
+  if (!matchedSubagent) return undefined;
+
+  const action =
+    typeof input.action === 'string' && input.action.trim()
+      ? input.action.trim()
+      : EXECUTIONS_DEFAULT_ACTION;
+  return `${action}: ${targets.join(', ')}`;
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/shared/tools/executionsDisplay.ts]: Fixed at the shared display-model boundary in ff7e8c6272. Path targets now retain both the execution identity and the resource suffix, so both hosts render `view: reviewer/report` or `view: reviewer/workspace-files/review.md`. Added shared, CLI, and extension regressions.

**Comment by LionSR** [comment]
The resource-label regression is fixed at ff7e8c627278aa9582291a2877978dd79c5d25ed. The shared formatter preserves nested suffixes such as `reviewer/workspace-files/review.md`, with shared, CLI, and extension regressions. A concurrent guarded push was intentionally not forced over this stronger remote change. @codex review exact head ff7e8c627278aa9582291a2877978dd79c5d25ed

**Comment by LionSR** [comment]
Addressed the exact-head label finding at 6b7dd92200: the extension now projects the cleaned stream-tab label rather than the source-qualified agent registry key. The regression uses `builtInToolUse:reviewer` and expects the user-facing `reviewer`. Five focused shared/CLI/extension suites pass (153 tests), along with root/test/extension type checks, lint, formatting, and diff checks. @codex review exact head 6b7dd92200

File: packages/cli/src/chat/tui/panes/StaticConversationTranscript.tsx [hunk 6]
```diff
@@ -574,6 +594,7 @@ export function StaticConversationTranscript({
         <Box key={item.id} flexDirection="column">
           <StaticTranscriptItemContent
             colorEnabled={colorEnabled}
+            executionLabels={subagentExecutionLabels}
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/chat/tui/panes/StaticConversationTranscript.tsx:609]: Fixed in 50dd155f7c at the static-output ownership boundary. Static transcript identity now includes the content-stable execution-label projection as well as its owner. Any identity change remounts the static items and uses the existing known-origin clear-and-repaint path, so already-finalized rows are replaced instead of duplicated or left stale. The repaint callback was generalized from viewport-only naming, deleting the now-shallow viewport-change wrapper. Added a real patched-Ink regression that starts with a raw `/executions/.../report` row, delivers the label later, and verifies the final terminal frame contains exactly one `view: reviewer/report` row and no raw path.

**Comment by LionSR** [comment]
Static execution labels now refresh at 50dd155f7ca36381bc862773516b366ba502903e through the existing clear-and-repaint path: the static render key includes the label projection, and a key change clears scrollback before remounting, so the corrected row appears once rather than being appended as duplicate terminal history. The regression verifies the clear sequence, updated label/resource path, and one occurrence. @codex review exact head 50dd155f7ca36381bc862773516b366ba502903e

**Comment by LionSR** [comment]
The TUI harness now follows the renamed static repaint callback at 25a881b49ec5f07006dc47e8df448880cf92ade2, keeping PTY validation on the same wiring as production. @codex review exact head 25a881b49ec5f07006dc47e8df448880cf92ade2

**Comment by LionSR** [comment]
The static repaint regression's handle is now immutable at b88141e91b4d18ee81f9f146a484891f51ebe3d7. An explicit `inkRef` avoids both the lint failure and access to a render handle before initialization; the test assigns it immediately after mount and uses the optional repaint boundary. A concurrent guarded push was not forced over this stronger fix. @codex review exact head b88141e91b4d18ee81f9f146a484891f51ebe3d7

File: packages/extension/src/progressView/frontend/progressState.ts [hunk 7]
```diff
@@ -263,6 +264,21 @@ const activeIsToolUse$ = new Signal.Computed(() => {
...
 });
 
+/** Session-wide identity projection; ordinary log ticks do not rebuild it. */
+const subagentExecutionLabels$ = new Signal.Computed((): ExecutionLabels => {
+  const labels = new Map<string, string>();
+  for (const child of streamById$.get().values()) {
+    if (child.kind !== 'agent' || !child.parentStreamId || !child.executionId) {
+      continue;
+    }
+    const label = (child.agent ?? child.label).trim();
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/extension/src/progressView/frontend/progressState.ts]: Fixed in 6b7dd92200. The extension projection now uses the cleaned `child.label` that the stream tab displays, rather than the source-qualified registry key in `child.agent`. The regression deliberately sets `agent` to `builtInToolUse:reviewer` while asserting the execution row resolves to `reviewer`.
